README.md

```diff
@@ -102,25 +102,28 @@ Once the Terraform Backend project has been setup the following AWS backend reso
     * tdr-terraform-state-lock; 
     * tdr-state-lock-jenkins
   * IAM Groups: 
-    * tdr-terraform-administrators; 
-    * tdr-terraform-developers
+    * TDRTerraformAdministrators; 
+    * TDRTerraformDevelopers
   * IAM Policies: 
-    * *[env name]*_access_terraform_state; 
-    * *[env name]*_terraform_assume_role_policy; 
-    * read_terraform_state; 
-    * terraform_state_lock_access
+    * TDR[env name]AccessTerraformState; 
+    * TDR[env name]TerraformAssumeRolePolicy; 
+    * TDRReadTerraformState; 
+    * TDRTerraformStateLockAccess
     * TDRJenkinsNodePolicy
   * IAM Roles:
     * TDRJenkinsNodeRole 
   
 In the TDR AWS environment accounts the following AWS resources should be available in each of the AWS accounts:
   * IAM Roles: 
-    * *[env name]*-terraform-role
-    * TDRJenkinsECSUpdateRoleIntg-*[env name]*
+    * TDRTerraformRole[env name]
+    * TDRJenkinsECSUpdateRole-*[env name]*
   
   * IAM Policies:
-    * s3-terraform-policy-*[env name]*
-    * TDRJenkinsUpdateECSIntg-*[env name]*
+    * TDRFrontendTerraform[env name]*-a
+    * TDRFrontendTerraform[env name]-b
+    * TDRKeycloakTerraform[env name]-a
+    * TDRKeycloakTerraform[env name]-b
```

Comment:
I think I'm missing something fundamental here - why does this project need to know about services like the frontend and Keycloak? I thought the purpose of the Terraform scripts in this repo was just to create an S3 bucket and Dynamo table to store the main Terraform state?

If we add a new service, will we have to create more IAM roles here as well as in [tdr-terraform-environments](https://github.com/nationalarchives/tdr-terraform-environments)? That seems quite complicated - I'd have expected to just have to edit the other repo, not this one, especially since the README for this repo says "The Terraform scripts are to be used as a one off operation".

Sorry if I'm just missing a lot of context! :)

Reply:
The original idea was for it to create the Terraform "backend". However, to get the cross account terraform to work a role needs to be created in each of the environments that has the necessary permissions to create the AWS resources. These roles can then be assumed by a user in the management account to run the Terraform for each environment from the management account. Rather than "handcraft" these roles and permissions, it is safer to create them using Terraform as well. Yes it is complicated at the moment, but once development is complete this shouldn't be needed again.

Reply:
Thanks. It'd be good to have a chat about this with you and Sam when we're all in the office. We'll be gradually adding services throughout Beta and hopefully beyond, so it would be good to have a Terraform setup that's easy to update. Have you considered making the policies more general, so they allow you to create the kinds of resources we're using (e.g. ECS services, S3 buckets) so the IAM policies are less work to manage?

Reply:
Sure. Having more generic policies was something we considered, but there is a limit (10 I think) of the number of policies that can be attached to a role. So there is a risk that we can't add additional policies in the future. It's hard to say what is the best approach here.

Reply:
👍

Reply:
We've discussed this in person, and we're going to go with the current approach and keep an eye on how hard it is to add new services.

We might also need to rethink our approach if we hit the IAM limits for policy size and policies-per-role.